lib/active_model_serializers/adapter/base.rb

```diff
@@ -51,6 +51,20 @@ def include_meta(json)
         json[meta_key] = meta if meta
         json
       end
+
+      def translate_key_casing!(value, serialization_context)
```

Comment:
thoughts on writing these transforms in the style of active support inflections? https://github.com/rails-api/active_model_serializers/pull/1566

at least for the sake of examples in comments, 

maybe the key transforms could be in their own module?

```ruby
module WhateverNamespace::KeyTransforms
  module_function
  # transforms keys to UpperCamelCase or PascalCase (this is what through me down this train of thought in the first place, though, this could be because I've been in non-ruby land at work)
  #
  # @example:
  #    "some_key" => "SomeKey",
  # @see: active_support/inflector#camelize
  def camel(hash)
    hash.deep_transform_keys! { |key| key.to_s.camelize.to_sym }
  end
  ...
end
```

and then this method would become:
```ruby
return value unless serialization_context
transform = serialization_context.key_case
KeyTransforms.send(transform)
```